### DISCUSSION TITLE
Compile error on http-server example

### DISCUSSION THREAD

**Discussion by schilton** [cesanta/mongoose#2167] [Ask questions, get answers] [answered]
I am trying to compile the http-server example and it fails to establish a listener on port 80.  I have tried both suggested configuration statements with the same results and I have confirmed that port 80 is open in the firewall as you can see below.  Does anyone have any suggestions on what to try next?

# `*e9-admin@cellp-ubuntu-s-1vcpu-1gb-amd-nyc1-01*:*~/mongoose/examples/http-server*$ sudo ufw status
Status: active

To                         Action      From
—                         ———      ——
OpenSSH                    ALLOW       Anywhere                  
443                        ALLOW       Anywhere                  
80/tcp                     ALLOW       Anywhere                  
OpenSSH (v6)               ALLOW       Anywhere (v6)             
443 (v6)                   ALLOW       Anywhere (v6)             
80/tcp (v6)                ALLOW       Anywhere (v6)             
`

`
*e9-admin@cellp-ubuntu-s-1vcpu-1gb-amd-nyc1-01*:*~/mongoose/examples/http-server*$ make
cc main.c mongoose.c        -W -Wall -Wextra -g -I.   -DMG_HTTP_DIRLIST_TIME_FMT=“%Y/%m/%d %H:%M:%S” -DMG_ENABLE_LINES=1 -DMG_ENABLE_IPV6=1 -DMG_ENABLE_SSI=1  -o example                                    
./example                    
263044 1 sock.c:225:mg_open_listener    bind: 13
263044 1 net.c:173:mg_listen            Failed: http://0.0.0.0:80, errno 13
263044 1 main.c:89:main                 Cannot listen on http://0.0.0.0:80. Use http://ADDR:PORT or :PORT
make: *** [Makefile:20: all] Error 1

`
`
*e9-admin@cellp-ubuntu-s-1vcpu-1gb-amd-nyc1-01*:*~/mongoose/examples/http-server*$ make
cc main.c mongoose.c        -W -Wall -Wextra -g -I.   -DMG_HTTP_DIRLIST_TIME_FMT=“%Y/%m/%d %H:%M:%S” -DMG_ENABLE_LINES=1 -DMG_ENABLE_IPV6=1 -DMG_ENABLE_SSI=1  -o example                                    
./example                    
26e158 1 sock.c:225:mg_open_listener    bind: 13
26e158 1 net.c:173:mg_listen            Failed: :80, errno 13
26e158 1 main.c:89:main                 Cannot listen on :80. Use http://ADDR:PORT or :PORT
make: *** [Makefile:20: all] Error 1
`

**Comment by cpq** [comment, marked as answer]
@schilton port 80 is a privileged port, you need to be root to open it. Either become root, or use ports > 1024.

**Comment by schilton** [comment]
Thanks Sergey, I forgot about that.

[image]

Scott Chilton
e: ***@***.******@***.***) t:[903.521.3100](tel:209035213100) f:[877.532.7413](tel:208775327413)

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------
